graph/src/data/graphql/query_hash.rs

```diff
@@ -0,0 +1,455 @@
+use crate::prelude::q;
+use graphql_parser::query::{
+    Definition, Directive, Document, Field, FragmentDefinition, Mutation, Number,
+    OperationDefinition, Query, Selection, SelectionSet, Subscription, Text, Value,
+    VariableDefinition,
+};
+use graphql_tools::ast::{OperationTransformer, Transformed, TransformedValue};
+use std::cmp::Ordering;
+use std::collections::{hash_map::DefaultHasher, BTreeMap};
+use std::hash::{Hash, Hasher};
+use std::marker::PhantomData;
+
+pub fn query_hash(query: &q::Document) -> u64 {
```

Comment:
This function should have a comment that describes how this hash works, similar to what's in the description of this PR